compiler/src/dmd/parse.d

```diff
@@ -4995,7 +5001,20 @@ class Parser(AST, Lexer = dmd.lexer.Lexer) : Lexer
         case TOK.delegate_:
             save = token.value;
             nextToken();
-            if (token.value == TOK.ref_)
+            if (token.value == TOK.auto_)
+            {
+                nextToken();
+                if (token.value == TOK.ref_)
+                {
+                    // function auto ref (parameters) { statements... }
+                    // delegate auto ref (parameters) { statements... }
+                    stc = STC.auto_ | STC.ref_;
+                    nextToken();
+                }
+                else
+                    error("`auto` can only be used as part of `auto ref` for function literal return values");
```

Comment:
Should we error here ? `function auto` is just like `function`. So either don't error or add an `errorSupplemental` telling the user to remove `auto` / `auto` is already the default behavior ?

Reply:
`function auto` is currently an error. There's an argument to be made for allowing it, but IMO it is out of scope for this PR.